File: ContactBookCQRS.Domain/Models/Category.cs
using ContactBookCQRS.Domain.Core.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ContactBookCQRS.Domain.Models
{
    public class Category : Entity
    {
        public Category(Guid id, Guid contactBookId, string name)
        {
            Id = id;
            ContactBookId = contactBookId;
            Name = name;
        }

        public string Name { get; private set; }
        public ICollection<Contact> Contacts { get; set; }
        public Guid ContactBookId { get; private set; }

        // Empty constructor for EF
        protected Category() { }
    }
}

File: ContactBookCQRS.WebApp/ClientApp/src/app/core/models/Category.ts
import { Contact } from './Contact';

export class Category {
  id: string;
  contactBookId: string;
  name: string;
  contacts: Contact[];

  constructor(contactBookId: string,
    name:string) {
    this.contactBookId = contactBookId;
    this.name = name;
  }
}
